test-validator/src/lib.rs

```diff
@@ -740,6 +733,18 @@ impl TestValidatorGenesis {
             .unwrap_or_else(|err| panic!("Test validator failed to start: {err}"))
     }
 
+    /// Start a test validator with the address of the mint account that will receive tokens
+    /// created at genesis (async version).
+    pub async fn start_async_with_mint_address(
+        &self,
+        mint_address: Pubkey,
+        socket_addr_space: SocketAddrSpace,
+    ) -> Result<TestValidator, Box<dyn std::error::Error>> {
+        let test_validator = TestValidator::start(mint_address, self, socket_addr_space, None)?;
+        test_validator.wait_for_nonzero_fees().await;
+        Ok(test_validator)
+    }
```

Comment:
This function isn't checking for deployed programs as `start_async_with_socket_addr_space` is -- is that intended?